packages/inspector/src/lib/data-layer/host/rpc-methods.ts

```diff
@@ -402,5 +407,28 @@ export async function initRpcMethods(
         throw new Error(`Custom asset with id ${assetId} not found`);
       });
     },
+
+    async getInspectorUIState() {
+      const InspectorUIState = engine.getComponentOrNull(EditorComponentNames.InspectorUIState);
+
+      if (InspectorUIState && InspectorUIState.has(engine.RootEntity)) {
+        const state = InspectorUIState.get(engine.RootEntity) as InspectorUIStateMessage;
+        return state;
+      }
+
+      // Return default state if component doesn't exist
+      return { sceneInfoPanelVisible: undefined };
+    },
+
+    async setInspectorUIState(req) {
+      const InspectorUIState = engine.getComponentOrNull(
+        EditorComponentNames.InspectorUIState,
+      ) as LastWriteWinElementSetComponentDefinition<unknown>;
+      if (InspectorUIState) {
+        InspectorUIState.createOrReplace(engine.RootEntity, req);
+        await compositeProvider.saveComposite(true);
+      }
+      return {};
+    },
```

Comment:
are this necessary? since it's a component, ui state will be saved automatically